pkg/wal/processor/transformer/wal_transformer.go

```diff
@@ -62,6 +64,16 @@ func WithLogger(l loglib.Logger) Option {
 	}
 }
 
+func WithValidator(validator func(*Transformer) error) Option {
```

Comment:
The validator function doesn't need to have the full transformer as parameter, just the transformer map, so we can pass that only, since it will be used by the transformer internally, and it will have access to it even though it's unexported (you can export the `columnTransfomers` type for that).

`func(map[string]ColumnTransformers) error`